test/e2e/serial/tests/workload_unschedulable.go

```diff
@@ -676,8 +676,23 @@ var _ = Describe("[serial][disruptive][scheduler] numaresources workload unsched
 			err = fxt.Client.Create(context.TODO(), dp)
 			Expect(err).ToNot(HaveOccurred())
 
-			By(fmt.Sprintf("checking deployment pods have been scheduled with the topology aware scheduler %q ", schedulerName))
+			// although the deployment pods will be pending thus the deployment will not be counted as complete,
+			// we need to wait until all the replicas are created despite their status before moving forward with the checks
+			dpKey := client.ObjectKeyFromObject(dp)
+			Eventually(func() bool {
+				err = fxt.Client.Get(context.TODO(), dpKey, dp)
+				Expect(err).ToNot(HaveOccurred())
+
+				if dp.Status.Replicas != replicas {
```

Comment:
I wonder if we should not just wrap the next checks in a `Eventually` loop, because this check is implicitely expecting the replicas will go up. Basically, we are assuming a positive test. This may be fine here, but I'm literally not sure it holds true in general

Reply:
What I think now is: is the reminder of the test (implicitely) expecting the replicas to be running, or is checking if the replica are actually all running part of the test itself?
IOW, are we:

- making an implicit expectation explicit?
- factoring a common expectation?
- something else?

Reply:
the test consists of two phases:
1. create a deployment that all its replicas keep on pending (because there are not enough resources)
2. update the requested resources under the deployment so that some of the replicas go running 
so the test is not completely negative, the tag [unsched] might be a bit misleading for this test.
Note though that we do expect the replicas will go up when in the first phase in the test, yet that doesn't mean they are running (running=ready=available), so the deployment pods are there but pending, which is the anticipated behavior.